scrc/dataset_creation/dataset_creator.py

```diff
@@ -652,6 +655,30 @@ def save_dataset(self, dataset: datasets.Dataset, labels: list, folder: Path,
         self.logger.info(f"Saved dataset files to {folder}")
         return splits
 
+    def clean_dataset(self, dataset):
+        # replace empty strings with nan so that they can be removed
+        self.logger.info(f"start cleaning")
+        empty_rows = []
+        def get_empty_rows(row):
+            if row["num_tokens"] <= self.minFeatureColLength:
+                empty_rows.append(row['__index_level_0__'])
+
+        for feature_col in self.get_feature_col_names():
+            dataset = dataset.rename_column(f"{feature_col}_num_tokens_bert", "num_tokens")
+            dataset.map(get_empty_rows)
+            dataset = dataset.rename_column("num_tokens", f"{feature_col}_num_tokens_bert")
+
+        duplicates = [number for number in empty_rows if empty_rows.count(number) == len(self.get_feature_col_names())]
+        non_empty_rows = [i for i in range(len(dataset)) if i not in duplicates]
+        dataset = dataset.select(non_empty_rows)
+
+        if 'html_url' in dataset.features.keys():
+            dataset = dataset.remove_columns(['html_url'])
+        if 'pdf_url' in dataset.features.keys():
+            dataset = dataset.remove_columns(['pdf_url'])
```

Comment:
why do you remove this? The urls can be very nice to have a quick look at the dataset. Maybe it would make sense though to just save one...

Reply:
I will remove it.